### ISSUE TITLE
java.lang.NoSuchMethodError: no static method on release only after moving to flutter beta

### ISSUE THREAD

**Issue by lucaszanella** [flutter/flutter#77406]
After moving to flutter beta:

```bash
lz@vm:~/stompai_pro/android/app$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 2.0.1, on Linux, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
```
I started getting `java.lang.NoSuchMethodError` on my app, when trying to load a static kotlin function from C++. This only happens in **release** mode:

03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036] No pending exception expected: java.lang.NoSuchMethodError: no static method `Lcom/myapp/flutter_app/JNI;.loadWavAsset(Ljava/lang/String;Ljava/lang/String;F)Z`

I tried adding these proguared rules to release mode:

```gradle
    buildTypes {
        release {
            minifyEnabled true
            useProguard true
            shrinkResources true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }

-keep public class com.myapp.flutter_app.JNIKt
-keep public class com.myapp.flutter_app.JNI
-keep public class com.myapp.flutter_app.MainActivity
```

but it won't help.

Kotlin file:

I'm getting a java.lang.NoSuchMethodError: no static method for a Kotlin function when loading it from C++, but it only happens when I run the Flutter RELEASE. In debug, it works without any errors.

```bash
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036] No pending exception expected: java.lang.NoSuchMethodError: no static method "Lcom/myapp/flutter_app/JNI;.loadWavAsset(Ljava/lang/String;Ljava/lang/String;F)Z"
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036]   at java.lang.String java.lang.Runtime.nativeLoad(java.lang.String, java.lang.ClassLoader, java.lang.String) (Runtime.java:-2)
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036]   at java.lang.String java.lang.Runtime.doLoad(java.lang.String, java.lang.ClassLoader) (Runtime.java:1099)
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036]   at void java.lang.Runtime.loadLibrary0(java.lang.ClassLoader, java.lang.String) (Runtime.java:1014)
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036]   at void java.lang.System.loadLibrary(java.lang.String) (System.java:1657)
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036]   at void com.myapp.flutter_app.MainActivity.<init>() (:-1)
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036]   at java.lang.Object java.lang.Class.newInstance() (Class.java:-2)
03-05 15:40:16.174  8799  8799 F zygote64: thread.cc:2036]   at android.app.Activity android.app.Instrumentation.newActivity(java.lang.ClassLoader, java.lang.String, android.content.Intent) (Instrumentation.java:1195)
```
Here's the kotlin file:

```kt
package com.myapp.flutter_app

import android.content.Context
//... other imports

class JNI(val _context: Context) {


    companion object {
        val TAG: String = "JNI.kt";

        lateinit var context: Context;

        @JvmStatic
        public fun loadWavAsset(player: String, assetPath: String, pan: Float): Boolean {

```

Maybe flutter beta does something different for proguard?

**Comment by TahaTesser** [comment]
Hi @lucaszanella 
Thanks for filing the issue, can you please provide a minimal reproducible code sample project?
Thank you

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
